src/basic/user-util.c

```diff
@@ -450,6 +451,59 @@ int in_group(const char *name) {
         return in_gid(gid);
 }
 
+size_t merge_gid_lists(
+        const gid_t *list1,
+        size_t size1,
+        const gid_t *list2,
+        size_t size2,
+        gid_t *result) {
```

Comment:
Since the kernel specifies the number of items in those lists as `int`, we should too. So please don't use `size_t` here.